lib/utils/ember.js

```diff
@@ -142,24 +149,42 @@ function isReopenObject(node) {
   return node.callee.property && node.callee.property.name === 'reopen';
 }
 
-function isEmberComponent(node, filePath) {
-  return isEmberCoreModule(node, 'Component', filePath);
+function isEmberCoreModule(context, node, moduleName) {
```

Comment:
Can we avoid passing the entire `context` object in? That's a huge object and I'd rather pass in only the specific data we need.

Reply:
Hey @bmish, we could (I'd prefer to) but it'd mean refactoring the `getImportModule` function, which is also being renamed in [the PR you mentioned before](https://github.com/ember-cli/eslint-plugin-ember/pull/548/files#diff-6086526e70749e816db18ecad72440ebR211) since that functions asks for `context`. I'm happy to refactor it here, but there will be more conflicts with that PR. 

You still think we should do it here? Or follow up with a PR after both of these have been merged, to combine them more easily?

Reply:
I just merged the other PR in case that helps. Can you rebase onto the latest master to pick up those changes?

I also noticed that the usage of `context` is for `getAncestors()` which makes sense. I'm not sure offhand if there's a better way to write it. So your change looks good to me and I can merge it after you rebase.

Reply:
Awesome, will do that now, thank you!

Sounds good

Reply:
I would have loved to avoid needing that `context` object in `getImportModule` (there are no real utilities for testing it from ESLint; you have to just write a mock yourself) but I think that using `getAncestors` is probably the "safest" way to walk back to `Program` node.

You can also walk back up the `.parent` tree, but `getAnvestors()[0]` might be faster than that -- I'm honestly not sure.

Reply:
Yeah, I think personally that I'd rather just pass in the whole `context` object -- the API is documented pretty well, at least!

I was really surprised that even within the ESLint repo it self, I didn't see any test helpers around this stuff that we could use. I think writing our own (for this repo, or if it's really good, something separate for other ESLint plugins' test suites to use) is probably our best bet!

Reply:
Yeah `context.getAnvestors()` sounds like the safest bet here.